Frameworks/UIKit/UIView.mm

```diff
@@ -178,12 +201,14 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
             [g_curGesturesDict setObject:arr forKey:gestureClass];
             [arr release];
         }
+
         [arr addObject:curgesture];
     }
 
-    for (int i = 0; i < viewDepth; i++) {
+    // sendTouch to gesture for state transiton
+    BOOL gestureOnGoing = NO; 
+    for (int i = 0; i < gestureCount; i++) {
```

Comment:
you should be able to do range-based for loops over both of these gesture arrays